torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -1833,6 +1864,56 @@ class IrParser {
           });
     }
 
+    { // LTC uses this op for softmax
+      auto ptr_op = getOperatorForLiteral(
+          "aten::_softmax(Tensor self, int dim, bool half_to_float) -> Tensor");
+      REGISTER_PARSE_RULE(
+          ptr_op,
+          {
+            MemoryFormat format;
+            std::list<Val*> list_val;
+            std::tie(format, list_val) = getConsistentValues(
+                MemoryFormat::Contiguous(),
+                value_map[node->inputs()[0]->unique()]);
+            auto input_t = list_val.front();
+            list_val.pop_front();
+            auto input = input_t->as<TensorView>();
+
+            auto dim_value = constant_as<int>(node->input(1));
+            TORCH_INTERNAL_ASSERT(
+                dim_value.has_value(), "dim in softmax is not valid");
+
+            auto output = softmax(input, dim_value.value());
+            value_map.emplace(node->output()->unique(), output);
+          },
+          [](const Node* node) -> bool {
+            if (node->inputs()[1]->node()->kind() != prim::Constant) {
+              return false;
+            }
+            if (node->inputs()[2]->node()->kind() == prim::Constant) {
```

Comment:
nitpitck: could simplify the logic down a little bit.

```
const auto half_to_float = constant_as<bool>(node->input(2));
auto input_tensor_type = node->input(0)->type()->cast<TensorType>();
// we don't support `half_to_float == false` or `half_to_float == true && input.dtype != half`
if (half_to_float.has_value() && (!half_to_float.value() || input_tensor_type->scalarType() != at::ScalarType::Half)) {
  return false;
}
return true; 
```

Reply:
I significantly fixed up this logic to make it simpler. I think there was a slight bug in your suggestion as we do want to accept `False` from the boolean.

Reply:
Oops, sorry about my buggy suggestion. :stuck_out_tongue: